operator/controllers/resources/role_controller_test.go

```diff
@@ -29,68 +34,203 @@ import (
 	kclient "sigs.k8s.io/controller-runtime/pkg/client"
 
 	"github.com/gravitational/teleport/api/types"
-	"github.com/gravitational/teleport/lib/auth"
 	resourcesv5 "github.com/gravitational/teleport/operator/apis/resources/v5"
 )
 
+// When I create or delete a TeleportRole CR in Kubernetes,
+// the corresponding TeleportRole must be created/deleted in Teleport.
 func TestRoleCreation(t *testing.T) {
 	ctx := context.Background()
-
-	teleportServer, operatorName := defaultTeleportServiceConfig(t)
-
-	require.NoError(t, teleportServer.Start())
-
-	tClient := clientForTeleport(t, teleportServer, operatorName)
-	k8sClient := startKubernetesOperator(t, tClient)
-
-	ns := createNamespaceForTest(t, k8sClient)
+	setup := setupKubernetesAndTeleport(t, ctx)
```

Comment:
Should we free resources like the auth client and k8s client?

Maybe add a Close method to setup so we can immediately follow this with `defer setup.Close()`?

Same for other usages.

Reply:
I think this one is still an issue.

Reply:
kube client does not seem to offer a close/cleanup method (or at least I was not able to find one).

I added teleport client & server cleanup in `setupKubernetesAndTeleport`.